crates/miden-objects/src/account/builder/mod.rs

```diff
@@ -243,6 +270,11 @@ mod tests {
               push.4.4 add eq.8
             end
           ";
+    const AUTH_CODE: &str = "
+            export.auth
+              push.1 assert
+            end
+          ";
```

Comment:
I think for most tests we'll need a `AccountAuthMockComponent` (analogously to `AccountMockComponent`) that will simply increment the nonce of the account, maybe it's worth adding this here already? I see similar code being added elsewhere in this PR. Maybe we can deduplicate it by adding this component in `crates/miden-objects/src/testing/account_component.rs` from where it should be accessible to all relevant test code. It might also replace `MockComponent` used in `Auth::Mock.build_component()`.

Reply:
I replaced this with the pre-defined `NoopAuthComponent` in [93aeb6f](https://github.com/0xMiden/miden-base/pull/1480/commits/93aeb6fa6dde089df15d65a886ecc643e4240dfd)